metrics/metrics-api/src/main/java/org/apache/james/metrics/api/MetricFactory.java

```diff
@@ -37,6 +39,11 @@
         }
     }
 
+    default <T> Mono<T> runPublishingTimerMetric(String name, Mono<T> mono) {
```

Comment:
Is `default <T> CompletableFuture<T> runPublishingTimerMetric` declared below still used somewhere else in the code or not? If not (as the goal as far I understand was to convert it to use Mono and get rid of Futures), can delete it perhaps?